File: server-scala-play-server/app/api/CSVImportApi.scala
package api

import play.api.libs.json._
import model.BigDecimal
import model.CsvImportResponse
import model.SirqulResponse
import play.api.libs.Files.TemporaryFile

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
trait CSVImportApi {
  /**
    * Detail Status
    * @param accountId the id of the logged in user
    * @param batchId the id of the batch
    * @param responseGroup The group of categories to return: SUMMARY, DETAILS, ERRORS, OR ALL
    * @param start the start of the pagination
    * @param limit the limit of the pagination
    */
  def getStatusCSV(version: BigDecimal, accountId: Long, batchId: Long, responseGroup: String, start: Long, limit: Long): SirqulResponse

  /**
    * Search Status
    * Retrieves batches for a user.
    * @param accountId the id of the account
    * @param start the start of the pagination
    * @param limit the limit of the pagination
    */
  def listStatusCSV(version: BigDecimal, accountId: Long, start: Int, limit: Int): CsvImportResponse

  /**
    * Batch Status
    * Checks status of batch upload.
    * @param accountId the id of the account
    * @param batchId the id of the batch to get its status
    */
  def statusCSV(version: BigDecimal, accountId: Long, batchId: Long): CsvImportResponse

  /**
    * Upload CSV
    * Uploads a CSV import file.
    * @param accountId the id of the account
    * @param uploadType the upload type: OFFERS, RETAILERS, RETAILERLOCATIONS, CATEGORIES, OR FILTERS
    * @param importFile the import file to reference
    * @param fileFormat the format of the file
    * @param appKey the application key
    */
  def uploadCSV(version: BigDecimal, accountId: Long, uploadType: String, importFile: TemporaryFile, fileFormat: String, appKey: Option[String]): CsvImportResponse
}

File: server-ruby-sinatra/api/csv_import_api.rb
require 'json'


MyApp.add_route('GET', '/api/{version}/csvimport/batch/status/details', {
  "resourcePath" => "/CSVImport",
  "summary" => "Detail Status",
  "nickname" => "get_status_csv",
  "responseClass" => "SirqulResponse",
  "endpoint" => "/api/{version}/csvimport/batch/status/details",
  "notes" => "",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "the id of the logged in user",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "batch_id",
      "description" => "the id of the batch",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "response_group",
      "description" => "The group of categories to return: SUMMARY, DETAILS, ERRORS, OR ALL",
      "dataType" => "String",
      "allowableValues" => "[SUMMARY, DETAILS, ERRORS, ALL]",
      "paramType" => "query",
    },
    {
      "name" => "start",
      "description" => "the start of the pagination",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "limit",
      "description" => "the limit of the pagination",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/{version}/csvimport/batch/list', {
  "resourcePath" => "/CSVImport",
  "summary" => "Search Status",
  "nickname" => "list_status_csv",
  "responseClass" => "CsvImportResponse",
  "endpoint" => "/api/{version}/csvimport/batch/list",
  "notes" => "Retrieves batches for a user.",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "the id of the account",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "start",
      "description" => "the start of the pagination",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "limit",
      "description" => "the limit of the pagination",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/{version}/csvimport/batch/status', {
  "resourcePath" => "/CSVImport",
  "summary" => "Batch Status",
  "nickname" => "status_csv",
  "responseClass" => "CsvImportResponse",
  "endpoint" => "/api/{version}/csvimport/batch/status",
  "notes" => "Checks status of batch upload.",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "the id of the account",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "batch_id",
      "description" => "the id of the batch to get its status",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('POST', '/api/{version}/csvimport/upload', {
  "resourcePath" => "/CSVImport",
  "summary" => "Upload CSV",
  "nickname" => "upload_csv",
  "responseClass" => "CsvImportResponse",
  "endpoint" => "/api/{version}/csvimport/upload",
  "notes" => "Uploads a CSV import file.",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "the id of the account",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "upload_type",
      "description" => "the upload type: OFFERS, RETAILERS, RETAILERLOCATIONS, CATEGORIES, OR FILTERS",
      "dataType" => "String",
      "allowableValues" => "[OFFERS, RETAILERS, RETAILERLOCATIONS, CATEGORIES, FILTERS]",
      "paramType" => "query",
    },
    {
      "name" => "import_file",
      "description" => "the import file to reference",
      "dataType" => "File",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "app_key",
      "description" => "the application key",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "file_format",
      "description" => "the format of the file",
      "dataType" => "String",
      "allowableValues" => "[EXCEL, MYSQL, RFC4180, TDF]",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end

